src/io.rs

```diff
@@ -384,13 +413,13 @@ impl ZkIo {
             watch: None,
         }
     }
-
+    
     async fn write_data(&mut self) {
         if let Some(tx) = self.sock_tx.as_mut() {
             while let Some(request) = self.buffer.pop_front() {
                 trace!("Writing data: {:?}", request.opcode);
-                match tx.write_all(request.data.chunk()).await {
-                    Ok(()) => {
+                match timeout(self.ping_timeout_duration, tx.write_all(request.data.chunk())).await {
```

Comment:
Can you flatten the result, so we don't have Result<Result<>>?

Reply:
The original code was fine, I started to think it was blocking here